internal/controllers/v1beta1/gateway/gateway.go

```diff
@@ -81,11 +83,14 @@ func (c *GatewayController) SetupWithManager(ctx context.Context, mgr manager.Ma
 
 	informer := istioInformerFactory.Networking().V1beta1().Gateways().Informer()
 	if c.gatewayLookupCache != nil {
-		informer.AddEventHandler(k8scache.ResourceEventHandlerFuncs{
+		_, err := informer.AddEventHandler(k8scache.ResourceEventHandlerFuncs{
 			AddFunc:    c.gatewayLookupCache.AddFunc,
 			UpdateFunc: c.gatewayLookupCache.UpdateFunc,
 			DeleteFunc: c.gatewayLookupCache.DeleteFunc,
 		})
+		if err != nil {
+			log.Error(err, "error adding event handler to gateway informer")
```

Comment:
Since these now return an error; and this is in the setup; the behavior should be to `return err` and fail the startup

Reply:
added